sgt-back-end/server.js

```diff
@@ -0,0 +1,108 @@
+import express from 'express';
+import pg from 'pg';
+import ClientError from './client-error.js';
+
+const app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `select * from "grades"`;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (error) {
+    next(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      throw new ClientError(400, 'GradeID must be a positive integer');
```

Comment:
This is the right idea, but it won't work unless you mount the error handling middleware. You need to copy the error-middleware.ts file from express-error-handling and `use` it at the bottom of all your routes.

But that requires setting up your project to run TypeScript or converting those files to be straight JS. Sorry about that! For now, don't worry about it. For the next cohort I've put JS versions of those files in the starter files so they'll be able to do this more easily.